### PR TITLE
Cisco Stack - added option to ignore if ring is not redundant

### PR DESCRIPTION
**Pull request by da-friedl**
Hi,

The current release will exit the script if the stack ring is not redundant. We have some cases at our customer sites where the stack ring is not wanted to be redundant, but we want to know the all the information about the non-redundant stack (reason may be due to costs of a second cable, the potential risk is accepted by the customer).

Best regards

Daniel

### CODE REVIEW

**Comment by garnier-quentin** [comment]
Thanks for the PR. We do the review soon.

**Comment by Sims24** [comment]
Hi @da-friedl thanks for your contrib. 

Few things to change please, I'll add my comments directly in the review

File: centreon/common/cisco/standard/snmp/mode/stack.pm [hunk 1]
```diff
@@ -229,17 +230,22 @@ my $oid_cswRingRedundant = '.1.3.6.1.4.1.9.9.500.1.1.3.0';
...
-    $self->{global} = { waiting => 0, progressing => 0, added => 0, ready => 0, sdmMismatch => 0, 
-        verMismatch => 0, featureMismatch => 0, newMasterInit => 0, provisioned => 0, 
+    $self->{global} = { waiting => 0, progressing => 0, added => 0, ready => 0, sdmMismatch => 0,
+        verMismatch => 0, featureMismatch => 0, newMasterInit => 0, provisioned => 0,
         invalid => 0, removed => 0 };
     $self->{members} = {};
 
     my $redundant = $self->{snmp}->get_leef(oids => [ $oid_cswRingRedundant ], nothing_quit => 1);
 
     if ($redundant->{$oid_cswRingRedundant} != 1) {
```
Comment (Sims24) [inline centreon/common/cisco/standard/snmp/mode/stack.pm]: Now you have a binary option mechanism, you can just transform this loop in something more readable 

```suggestion
    if (($redundant->{$oid_cswRingRedundant} != 1) && (defined($self->{$self->{option_results}->{ignore_cswringredundant)) {
```

You "save" an if loop :)

File: centreon/common/cisco/standard/snmp/mode/stack.pm [hunk 2]
```diff
@@ -229,17 +230,22 @@ my $oid_cswRingRedundant = '.1.3.6.1.4.1.9.9.500.1.1.3.0';
...
         $self->{output}->add_option_msg(short_msg => "Stack ring is not redundant");
-        $self->{output}->option_exit();
+
+        if ($self->{ignore_cswringredundant} eq ''){
```
Comment (Sims24) [inline centreon/common/cisco/standard/snmp/mode/stack.pm]: See previous comment ;)

**Comment by da-friedl** [comment]
see #1503

**Pull request closed without merging**